### PR TITLE
Declarative over imperative assertions

### PR DESCRIPTION
**Pull request by the-undefined**
I found it difficult following along with what the assertions were about. This is an attempt to tackle that using:
- a verbose test api
- removing magic number from assertions

What do you think @lee-jon does this improve or hamper readability?

### CODE REVIEW

File: gilded_rose_spec.rb [hunk 1]
```diff
@@ -1,206 +1,215 @@
...
+          to change{ item.quality }.
+          by(1)
       end
 
       it "should not increase beyond 50" do
-        51.times { @gildedrose.update_quality }
-
-        expect(@item.quality).to_not be > 50
+        expect{
+          52.times { gilded_rose.update_quality }
```
Comment (szalansky) [inline gilded_rose_spec.rb]: why did it increase?

File: gilded_rose_spec.rb [hunk 2]
```diff
@@ -1,206 +1,215 @@
...
 require "rspec"
 
 describe GildedRose do
+  let(:gilded_rose) { GildedRose.new(item) }
+  let(:zero_quality) { 0 }
+  let(:max_quality) { 50 }
+  let(:quality) { rand(max_quality) + 1 }
+
+  let(:zero_sell_in) { 0 }
+  let(:sell_in) { rand(50) + 1 }
```
Comment (yggie) [inline gilded_rose_spec.rb]: I personally don’t like the idea of using random numbers in tests, any reason to use this?

Comment (the-undefined) [inline gilded_rose_spec.rb]: It was a good refactoring aid to remove relliance on specific number assertions, but I agree this should be taken out now, it just represents a positive integer. 

We wouldn't want a test suite that has the opportunity to hide edge cases via it's randomness - was that your thinking too?

File: gilded_rose_spec.rb (hunk 1 above)
Comment (the-undefined) [inline gilded_rose_spec.rb]: It's in the test description, but there are still some magic numbers going on. I could signify the number like:

``` ruby
expect {
  (max_quality + 1).times { gilded_rose.update_quality }
}.to change{ item.quality }.
  to(max_quality)
```

But I also have a disjointed test description and assertion still. I'll update both.

File: gilded_rose_spec.rb (hunk 2 above)
Comment (yggie) [inline gilded_rose_spec.rb]: yes that was my thinking, if there are edge cases, they should be explicitly handled in tests, not implicitly by chance
